EXOSIMS/SurveySimulation/multiSS.py

```diff
@@ -0,0 +1,904 @@
+        self.counter_3 = counter_3
+        # Add to outspec
+        self._outspec["coeffs"] = coeffs
+
+        # self.seed = specs.get("seed")
+        # self._outspec["seed"] = self.seed
+
+        # normalize coefficients
+        coeffs = np.array(coeffs)
+        # coeffs = coeffs / np.linalg.norm(coeffs)
+
+        # initialize the second target star
+        self.second_target = None
+        # to handle first two target case
+        self.counter_2 = None
+        self.oldInd = None
+        self.slewTimes_2 = None
+
+    def run_sim(self):
+        """Performs the survey simulation"""
+
+        OS = self.OpticalSystem
+        TL = self.TargetList
+        SU = self.SimulatedUniverse
+        Obs = self.Observatory
+        TK = self.TimeKeeping
+
+        # TODO: start using this self.currentSep
+        # set occulter separation if haveOcculter
+        if OS.haveOcculter:
+            self.currentSep = Obs.occulterSep
+
+        # choose observing modes selected for detection (default marked with a flag)
+        allModes = OS.observingModes
+        det_mode = list(filter(lambda mode: mode["detectionMode"], allModes))[0]
+        # and for characterization (default is first spectro/IFS mode)
+        spectroModes = list(
+            filter(lambda mode: "spec" in mode["inst"]["name"], allModes)
+        )
+        if np.any(spectroModes):
+            char_mode = spectroModes[0]
+        # if no spectro mode, default char mode is first observing mode
+        else:
+            char_mode = allModes[0]
+
+        # begin Survey, and loop until mission is finished
+        log_begin = "OB%s: survey beginning." % (TK.OBnumber)
+        self.logger.info(log_begin)
+        self.vprint(log_begin)
+        t0 = time.time()
+        sInd = None
+        ObsNum = 0
+        while not TK.mission_is_over(OS, Obs, det_mode):
+
+            # acquire the NEXT TARGET star index and create DRM
+            old_sInd = sInd  # used to save sInd if returned sInd is None
+            DRM, sInd, det_intTime, waitTime = self.next_target(sInd, det_mode)
+
+            # some new logic, subject to changes (change to 1800 for 5Y)
+            if (
+                TK.currentTimeNorm.copy().value
+                + waitTime.value
+                + Obs.settlingTime.value
+                + det_intTime.value
+            ) > 1810:
+                dtsim = (time.time() - t0) * u.s
+                log_end = (
+                    "Mission complete: no more time available.\n"
+                    + "Simulation duration: %s.\n" % dtsim.astype("int")
+                    + "Results stored in SurveySimulation.DRM"
+                    "(Design Reference Mission)."
+                )
+                self.logger.info(log_end)
+                self.vprint(log_end)
+                break
+
+            if sInd is not None:
+                ObsNum += (
+                    1  # we're making an observation so increment observation number
```

Comment:
place comments above code unless it fits on same line without wrapping

Reply:
fixed